includes/class-telemetry.php

```
@@ -0,0 +1,648 @@
+				'scan_type'                 => $scan_type,
+				// New metrics (Phase 1 refinements).
+				'dom_size'                  => $resources['dom_size'],
+				'unminified_assets_count'   => $resources['unminified_count'],
+				'third_party_scripts_count' => $resources['third_party_count'],
+			);
+
+			set_transient( $transient_key, $result, HOUR_IN_SECONDS );
+			self::register_transient_key( $transient_key );
+
+			return $result;
+		}
```

PHPMD (2.15.0)

[warning] The method scan() has a Cyclomatic Complexity of 20. The configured cyclomatic complexity threshold is 10. (undefined)

(CyclomaticComplexity)